OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -132,6 +146,59 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
         }
     }
 
+    private void evaluateJavascriptForiFrame() {
+        if (!TextUtils.isEmpty(unit.getBlockId())) {
+            // execute js code to check an HTML block that contains an iframe
+            String javascript =
+                    "try {" +
+                            "    var top_div_list = document.querySelectorAll('div[data-usage-id=\"" + unit.getId() + "\"]');\n" +
+                            "    top_div_list.length == 1 && top_div_list[0].querySelectorAll(\"iframe\").length > 0;" +
+                            "} catch {" +
+                            "    false;" +
+                            "};";
+            authWebView.evaluateJavascript(javascript, value -> {
+                if (Boolean.parseBoolean(value)) {
+                    setupOpenInBrowserView();
+                } else {
+                    tvOpenBrowser.setVisibility(View.GONE);
+                }
+            });
+        }
+    }
+
+    private void setupOpenInBrowserView() {
+        tvOpenBrowser.setVisibility(View.VISIBLE);
+
+        String openInBrowserMessage = getString(R.string.open_in_browser_message) + " "
+                + getString(R.string.open_in_browser_text) + " " + "icon";
+        SpannableString openInBrowserSpan = new SpannableString(openInBrowserMessage);
+
+        ImageSpan openInNewIcon = new ImageSpan(getContext(), R.drawable.ic_open_in_new);
+        openInBrowserSpan.setSpan(openInNewIcon, openInBrowserMessage.indexOf("icon"),
```

Comment:
Can we use a meaningful constant here instead of `icon` ?

Reply:
`icon` string is supposed to replace with the drawable, it can be any string. Just used the `icon` string to make meanings that the actual icon should be placed here.

Reply:
Having a constant identifier will make sure that we are always passing the right value. :smile:

Reply:
done.